.github/workflows/python-app.yml

```
@@ -0,0 +1,42 @@
+name: Python application
+
+on:
+  push:
+    branches: [ main, develop ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        python-version: [ 3.9 ]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade -r requirements.txt
+      - name: Lint with flake8
+        run: |
+          flake8 . --count --show-source --statistics
+      - name: Lint with pylint
+        run: |
+          pylint src tests
+      - name: Test with pytest
+        run: |
+          pytest
```

I think the step used in Whist-Server are better.
```suggestion
    - name: Python Pytest with Coverage
      run: |
        python3 -m pytest --cov=server --cov-report=xml
    - name: Upload to codecoverage
      uses: codecov/codecov-action@v1
      with:
        file: ./coverage.xml
        fail_ci_if_error: true
        token: ${{ secrets.CODECOV_TOKEN }}
```
We should also use a separate workflow for pylint as on fail we would get any information about the tests.